test/tests/meshgenerators/block_deletion_generator/delete_interior_parents.i

```diff
@@ -0,0 +1,90 @@
+[Mesh]
+  final_generator = delete
+  [twoquad]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 3
+    ny = 4
+    xmax = 2
+  []
+  [subdomain]
+    type = SubdomainBoundingBoxGenerator
+    input = twoquad
+    block_id = 1
+    bottom_left = '1 0 0'
+    top_right = '2 1 0'
+  []
+  [sideset2]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = subdomain
+    primary_block = 1
+    paired_block = 0
+    new_boundary = bar
+  []
+  [edge_elem2]
+    type = LowerDBlockFromSidesetGenerator
+    input = sideset2
+    sidesets = bar
+    new_block_id = 2
```

Comment:
Well, the lowerD block is, to the block_deletion_generator, just another block, the fact that someone made a new block with a new block id using the LowerDBlock generator shouldn't have any bearing. It doesn't make any sense that it wouldn't be able to delete the block, the selection of block is essentially arbitrary.

As long as it properly discerns between the right elements, there shouldn't be an issue, right? As far as I know that's what is happening.

Reply:
Let's say we have 
block 1, 2 higher D blocks
block 3 tied to elements in block 1, lower D, on sideset 1 between 1 & 2
block 4 tied to elements in block 2, lower D, on sideset 2 between 2 & 1

I d want the BlockDeletionGenerator to be able to:
- delete block 3 when deleting block 1 (this might be controversial actually)
- delete block 4 when deleting block 2 (same)
- delete block 3 or 4 only

Reply:
Do you want me to test for that behavior?

Reply:
I d like the test that deletes block 1 & 3 also contain block 4 and does not delete it.

You are pretty much there, you just need to get the MGs set up properly and fix the block ids created I think

Reply:
As far as i'm concerned, the most important thing is to properly document this (eg, with examples, similar to @GiudGiud 's above).  BUT, if i had my choice, i would not want to always delete block 3 when deleting block 1.  I don't see why you'd always want to do that, and in my situation you'd almost always want to not delete block 3.  I'd prefer to specify block = '1 3' if i wanted to delete both blocks.   But, as i said, the documentation is more important than the final functionality.

Reply:
The problem is that:
- if you specify '1 3' you already delete both blocks currently
- if you only specify 1, it seg faults because of elements missing their interior parent.
I agree documenting the behavior is important here

Reply:
Newest push deletes block 1 and 3 and doesnt delete block 4.